src/Configuration/Config.KeyPerFile/src/KeyPerFileConfigurationBuilderExtensions.cs

```diff
@@ -19,6 +19,18 @@ public static class KeyPerFileConfigurationBuilderExtensions
         /// <param name="optional">Whether the directory is optional.</param>
         /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
         public static IConfigurationBuilder AddKeyPerFile(this IConfigurationBuilder builder, string directoryPath, bool optional)
+            => builder.AddKeyPerFile(directoryPath, optional, reloadOnChange: false);
```

Comment:
question - do our other config providers do reloading by default? Looks like JSON does not: https://github.com/aspnet/Extensions/blob/master/src/Configuration/Config.Json/src/JsonConfigurationExtensions.cs#L38

Reply:
We should be consistent so it should be off by default.

Reply:
Yeah this is expensive and has some unexpected side effects so generally reload should be off by default unless people explicitly make it enabled